modules/bvl_feedback/php/module.class.inc

```diff
@@ -35,4 +47,80 @@ class Module extends \Module
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param string $type The type of widgets to get
+     * @param \User  $user The user widgets are being retrieved for.
+     *
+     * @return \LORIS\GUI\Widget[]
+     */
+    public function getWidgets(string $type, \User $user)
+    {
+        switch($type) {
+        case 'dashboard':
+            $factory = \NDB_Factory::singleton();
+            $baseURL = $factory->settings()->getBaseURL();
+            $DB      = $factory->database();
+
+            $last_login = $user->getLastLogin($DB);
+
+            $bvl_feedback      = $DB->pselect(
+                "SELECT fbt.Name, fbe.Testdate, fbe.Comment, fbth.FieldName,
+                fbth.CommentID, fbth.SessionID, fbth.CandID, fbth.Feedback_level
+                FROM feedback_bvl_entry fbe
+                JOIN feedback_bvl_thread fbth USING (FeedbackID)
+                JOIN feedback_bvl_type fbt USING (Feedback_type)
+                WHERE fbth.Status='opened' AND fbth.Active='Y'
+                ORDER BY Testdate DESC LIMIT 4",
+                array()
+            );
+            $frontend_feedback = array();
+            foreach ($bvl_feedback as $row) {
+                if (new \DateTime($row['Testdate']) > $last_login) {
+                    $row['new'] = true;
+                } else {
+                    $row['new'] = false;
+                }
+                if ($row['Feedback_level'] === 'profile') {
+                    $row['URL'] = '/' . $row['CandID'];
+                } else if ($row['Feedback_level'] === 'visit') {
+                    $row['URL'] = '/instrument_list/?candID='
+                        . $row['CandID'] . '&sessionID='
+                        . $row['SessionID'];
+                } else if ($row['Feedback_level'] === 'instrument') {
+                    $instrument = $DB->pselectOne(
+                        "SELECT Test_name from flag WHERE CommentID=:cid",
+                        array('cid' => $row['CommentID'])
+                    );
+                    if (!empty($instrument)) {
+                        $row['URL'] = '/instruments/'
+                            . $instrument . '/?candID='
+                            . $row['CandID'] . '&sessionID='
+                            . $row['SessionID'] . '&commentID='
+                            . $row['CommentID'];
+                    }
+                }
+                $frontend_feedback[] = $row;
+            }
+            return [
+                new \LORIS\dashboard\Widget(
+                    new \LORIS\dashboard\WidgetContent(
+                        "Behavioural Feedback Notifications",
+                        $this->renderTemplate(
+                            "dashboardwidget.tpl",
+                            [
+                                'baseURL'       => $baseURL,
+                                'notifications' => $frontend_feedback
+                            ],
+                        ),
+                        "",
```

Comment:
same

Reply:
I tried to run your PR locally, but I got all same error "syntax error, unexpected ')' in /var/www/Loris/modules/dashboard/php/module.class.inc on line 120"
modules/bvl_feedback/php/module.class.inc
modules/dashboard/php/dashboard.class.inc
modules/dashboard/php/module.class.inc
modules/document_repository/php/module.class.inc
modules/statistics/php/module.class.inc 
You seem like add a unnecessary comma in the functioins.

Reply:
What version of PHP are you using?

Reply:
7.2

Reply:
Travis in on php 7.3

Reply:
Trailing commas being optional in function calls was added to PHP in 7.3 (and Travis now only runs 7.3 and 7.4, so that's not why it's failing.. ), but I removed the comma anyways